modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvIterator.java

```diff
@@ -0,0 +1,281 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ * 
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.collect.io;
+
+import java.io.BufferedReader;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.PeekingIterator;
+import com.google.common.io.CharSource;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.Unchecked;
+
+/**
+ * Iterator over the rows of a CSV file.
+ * <p>
+ * Provides the ability to iterate over a CSV file together with the ability to parse it from a {@link CharSource}.
+ * The separator may be specified, allowing TSV files (tab-separated) and other similar formats to be parsed.
+ * See {@link CsvFile} for more details of the CSV format.
+ * <p>
+ * This class processes the CSV file row-by-row.
+ * To load the entire CSV file into memory, use {@link CsvFile}.
+ * <p>
+ * This class must be used in a try-with-resources block to ensure that the underlying CSV file is closed:
+ * <pre>
+ *  try (CsvIterable iterable = CsvIterable.of(source, true)) {
```

Comment:
There are quite a few references to `CsvIterable` in the Javadoc and in some message strings. Did the class have a late name change but the docs and strings didn't get updated?